src/model.py

```diff
@@ -0,0 +1,614 @@
+import math
+import os
+import numpy as np
+import einops
+
+import torch
+from torch import nn, Tensor
+import torch.nn.functional as F
+from torch.utils.data import dataset
+
+from torchvision.models.resnet import BasicBlock, ResNet, conv1x1
+
+
+def custom_weight_init(m, init_scaling):
+    if hasattr(m, "weight"):
+        m.weight.data *= init_scaling
+
+        if m.bias is not None:
+            m.bias.data *= init_scaling
+
+#Defining the convolutional neural network
+class LeNet5(nn.Module):
+    def __init__(self, num_classes=10):
+        super(LeNet5, self).__init__()
+        self.layer1 = nn.Sequential(
+            nn.Conv2d(1, 6, kernel_size=5, stride=1, padding=0),
+            nn.BatchNorm2d(6),
+            nn.ReLU(),
+            nn.MaxPool2d(kernel_size = 2, stride = 2))
+        self.layer2 = nn.Sequential(
+            nn.Conv2d(6, 16, kernel_size=5, stride=1, padding=0),
+            nn.BatchNorm2d(16),
+            nn.ReLU(),
+            nn.MaxPool2d(kernel_size = 2, stride = 2))
+        self.fc1 = nn.Linear(400, 120)
+        self.relu = nn.ReLU()
+        self.fc2 = nn.Linear(120, 84)
+        self.relu1 = nn.ReLU()
+        self.fc3 = nn.Linear(84, num_classes)
+        
+    def forward(self, x):
+        out = self.layer1(x)
+        out = self.layer2(out)
+        out = out.reshape(out.size(0), -1)
+        out = self.fc1(out)
+        out = self.relu(out)
+        out = self.fc2(out)
+        out = self.relu1(out)
+        out = self.fc3(out)
+        return out
+
+class MyBasicBlock(BasicBlock):
```

Comment:
Might be mistaken about this, but I don't think this class is used anywhere else? Can delete if that's the case